core/trino-main/src/main/java/io/trino/operator/output/SlicePositionsAppender.java

```diff
@@ -80,38 +80,43 @@ public void append(IntArrayList positions, Block block)
             return;
         }
         ensurePositionCapacity(positionCount + positions.size());
-        int[] positionArray = positions.elements();
-        int newByteCount = 0;
-        int[] lengths = new int[positions.size()];
-
-        if (block.mayHaveNull()) {
-            for (int i = 0; i < positions.size(); i++) {
-                int position = positionArray[i];
-                if (block.isNull(position)) {
-                    offsets[positionCount + i + 1] = offsets[positionCount + i];
-                    valueIsNull[positionCount + i] = true;
-                    hasNullValue = true;
+        if (block instanceof VariableWidthBlock) {
```

Comment:
`if (block instanceof VariableWidthBlock) {`
->
`if (block instanceof AbstractVariableWidthBlock) {`

Reply:
there is a possible performance impact if w have more than 1 type here. Also, with `VariableWidthBlock`, just CHA can provide optimal compilation here, even with C1 as there is only one type. With `AbstractVariableWidthBlock` there are always 2 (builder) so CHA cannot fully help here so jit has to rely on the call site profile

Reply:
talked offline, `AbstractVariableWidthBlock` may not suppoort one single Slice. I'm leaving `VariableWidthBlock`